### ISSUE TITLE
Incorrect production dependencies

### ISSUE THREAD

**Issue by not-sentient** [honeybadger-io/honeybadger-js#1288]
## What are the steps to reproduce this issue?
1. Require `@honeybadger-io/js` in your production dependencies.
2. Bundle your project for production with Webpack.
3. The final bundle (package-lock and node_modules/) will include `cosmiconfig`, `@types/lambda`, and `@types/express`.

## What happens?
This increases node_modules/ by at least 35MB as `cosmiconfig` transitively brings in `typescript` as well (a peer dependency).

## What were you expecting to happen?
`@honeybadger-io/js` should not bring in `@types/*`/`typescript` into production bundles.

## Any logs, error output, etc?
N/A

## Any other comments?
Why were `@types/express` and `@types/lambda` added to production dependencies? These should be devDependencies.

## What versions are you using?
`npm --> 9.6.7`
`node --> 18.17.1`

**Comment by subzero10** [comment]
Hey @not-sentient, thank you for submitting an issue!

`cosmiconfig` lists `typescript` as an optional dependency and the default behavior of package managers is to install optional dependencies. For this reason, since [@honeybadger-io/js@6.7.2](https://github.com/honeybadger-io/honeybadger-js/releases/tag/%40honeybadger-io%2Fjs%406.7.2), we have removed the dependency to `cosmiconfig`.

In regards to the typings dependencies, the reason we added them is because some users were facing [issues](https://github.com/honeybadger-io/honeybadger-js/issues/907) when using Honeybadger with Typescript. There are a few other solutions we could explore:
- Ask users to avoid type-checking dependencies by tuning Typescript configuration (such as `skipLibCheck`). We didn't go for this because it will not always work and any configuration applies to the whole project, not just the Honeybadger dependency.
- Maintain internally type definitions of _express_ and _aws lambda_. I personally didn't want to go this route because then we'd have to manually monitor the type definitions of these packages and update our own version.

**Comment by not-sentient** [comment]
Hey, thanks for the quick reply!

Regarding `cosmiconfig`, isn't still in the dependencies (both in the linked you provided and on `master`)? We're using 6.8.2 and we're seeing `cosmiconfig` in our production bundle, brought in along with `@types/express` and `@types/lambda`. The types packages are a small nuisance to have in production bundles but aren't nearly as bad as `cosmiconfig` and `typescript`. For Lambda environments this has significant performance and cost impacts.

For the types packages, it seems like a combination of peerDependencies and optional external type imports might work as per StackOverflow: 

https://stackoverflow.com/questions/54392809/how-do-i-handle-optional-peer-dependencies-when-publishing-a-typescript-package

e.g.:
```
// @ts-ignore
import type { Something } from "optional"
import type { SomethingElse } from "required"
```

**Comment by subzero10** [comment]
> Regarding cosmiconfig, isn't still in the dependencies (both in the linked you provided and on master)?

Oh shoot, you are absolutely right! It seems there's been a hiccup with some commits or I simply forgot to remove it from the `package.json`. I created #1289 that removes it. It should be released soon (probably today).

Regarding the _types_ packages, if you go through the SO post's comments, you will see that there is no perfect solution. In fact, we actually use the `import type` modifiers (see [here](https://github.com/honeybadger-io/honeybadger-js/blob/93b8d91ceaeda45c3907bde127d3d5e89a7af2ba/packages/js/examples/aws-lambda-typescript/app/handler.ts#L1)), but removing the dependency (and adding `//@ts-ignore`) will hinder developer experience as they won't be able to see type definitions for the lambda handlers. As you said, _types_ packages don't have an impact in performance and _usually_ have very little impact in bundles. If that's not the case, then we should consider alternative solutions.

We are also thinking that sometime in the future we could release an AWS Lambda (or a serverless) specific package instead of the universal package, which would make these dependencies only required for solutions that would actually use them.

**Comment by not-sentient** [comment]
Thank you! Removing `cosmiconfig` will be a fantastic improvement. As for the types, I think their impact is minimal and we can definitely live with them.

The AWS Lambda specific package sounds awesome. I think this issue be can closed whenever you see fit.

Thanks a lot!

**Comment by subzero10** [comment]
Hey @not-sentient, this should be fixed with `v6.8.3`!
